File: ParadigmaComp/15Questoes/python/questao14.py
#Singleton
class Configuracao:
    _instancia = None

    def __new__(cls):
        if cls._instancia is None:
            cls._instancia = super().__new__(cls)
        return cls._instancia

# Exemplo de uso
config1 = Configuracao()
config2 = Configuracao()

print(config1 == config2)  # True (mesma instância)

File: ParadigmaComp/15Questoes/golang/questao14.go
package main
//Singleton
import "fmt"

// Estrutura Singleton
type Configuracao struct{}

var instancia *Configuracao

func GetConfiguracao() *Configuracao {
	if instancia == nil {
		instancia = &Configuracao{}
	}
	return instancia
}

func main() {
	config1 := GetConfiguracao()
	config2 := GetConfiguracao()

	fmt.Println(config1 == config2)  // True (mesma instância)
}
